dmap.js

```diff
@@ -70,10 +58,18 @@ lib._slot = async (dmap, key) => {
 }
 
 lib.walk = async (dmap, path) => {
+    if (![':', '.'].includes(path.charAt(0))) {
+        path = ':' + path
+    }
+    if (path.includes('.')) {
+        need(path.lastIndexOf(":") < path.indexOf("."), `Encountered ':' in unlocked subpath`)
```

Comment:
There’s a reason this was done via the “context”, other runes might have other logic where you can’t make such a single global rule